administrator/components/com_scheduler/src/Model/TaskModel.php

```diff
@@ -280,6 +293,188 @@ public function getItem($pk = null)
+	 *                           task queue. (default: false)
+	 *                           4. `allowConcurrent`: If true, fetches even when another task is
+	 *                           running ('locked'). (default: false)
+	 *
+	 * @return ?\stdClass  Task entry as in the database.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws UndefinedOptionsException|InvalidOptionsException
+	 * @throws \RuntimeException
+	 */
+	public function getTask(array $options = []): ?\stdClass
+	{
+		$resolver = new OptionsResolver;
+
+		try
+		{
+			$this->configureTaskGetterOptions($resolver);
+		}
+		catch (\Exception $e)
+		{
+		}
+
+		try
+		{
+			$options = $resolver->resolve($options);
+		}
+		catch (\Exception $e)
+		{
+			if ($e instanceof UndefinedOptionsException || $e instanceof InvalidOptionsException)
+			{
+				throw $e;
+			}
+		}
+
+		$db  = $this->getDbo();
+		$now = Factory::getDate()->toSql();
+
+		// Get lock on the table to help with concurrency issues
+		$db->lockTable(self::TASK_TABLE);
+
+		$lockQuery = $db->getQuery(true);
+
+		// We'll attempt to get a lock on the task requested, if another task does not already have a pseudo-lock.
+		$lockQuery->update($db->quoteName(self::TASK_TABLE))
+			->set($db->quoteName('locked') . ' = :now1')
+			->bind(':now1', $now);
+
+		if (!$options['bypassScheduling'])
+		{
+			$lockQuery->where($db->quoteName('next_execution') . ' <= :now2')
+				->bind(':now2', $now);
+		}
+
+		if (!$options['allowConcurrent'])
+		{
+			$subQuery = $db->getQuery(true);
+
+			// Sub-query to get count of locked tasks.
+			$subQuery->from($db->quoteName(self::TASK_TABLE))
+				->select('COUNT(*)')
+				->where($db->quoteName('locked') . ' IS NOT NULL');
+
+			$lockQuery->where("($subQuery) = 0");
+		}
+
+		if ($options['allowDisabled'])
+		{
+			$lockQuery->whereIn($db->quoteName('state'), [0, 1]);
+		}
+		else
+		{
+			$lockQuery->where($db->quoteName('state') . ' = 1');
+		}
+
+		if ($options['id'] > 0)
+		{
+			$lockQuery->where($db->quoteName('id') . ' = :taskId')
+				->bind(':taskId', $options['id'], ParameterType::INTEGER);
+		}
+		else
+		{
+			$idQuery = $db->getQuery(true);
+			$idWrapperQuery = $db->getQuery(true);
+
+			// Get the id of the next task in the task queue
+			$idQuery->from($db->quoteName(self::TASK_TABLE))
+				->select('id')
+				->where($db->quoteName('state') . ' = 1')
+				->order($db->quoteName('priority') . ' DESC')
+				->order($db->quoteName('next_execution') . ' ASC')
+				->setLimit(1)
+				->alias($db->quoteName('nextTask'));
+
+			// Wrap the id query for compatibility
+			$idWrapperQuery->select('*')->from($idQuery);
+
+			// $lockQuery->join('INNER', $idQuery, $db->quoteName('#__scheduler_tasks.id') . ' = ' . $db->quoteName('nextTask.id'));
+			$lockQuery->where($db->quoteName('id') . " IN ($idWrapperQuery)");
+		}
+
+		try
+		{
+			// @todo Remove -- Dbg
+			$queryStr = $lockQuery->__toString();
+			$db->setQuery($lockQuery)->execute();
```

Comment:
Fails here with MySQL / MariaDB -- Table `<prfx>_scheduler_tasks` not locked with LOCK TABLES. Seems to work with PostgreSQL.

Reply:
resolved by avoiding subqueries